app/views/admin/applications/show.html.erb

```diff
@@ -0,0 +1,52 @@
+<h1> Admin Applications</h1> 
+<br>
+
+<h2>Name of Applicant: <%= @application.name %></h2>
+
+<section id="applicant-info">
+<p>
+  Applicant Address:
+  <%= @application.street_address %>,
+  <%= @application.city %>, <%= @application.state %>, <%= @application.zip_code %>
+</p>
+<p>Description of why the applicant says they'd be a good home for this pet(s): <%= @application.description %></p>
+</section>
+
+<br>
+
+<section id="pets-for-adoption">
+<h2>Pets for Adoption:</h2>
+<ul>
+  <% @application.pets.each do |pet| %>
+    <li><%= link_to pet.name, "/pets/#{pet.id}" %></li> 
+  <% end %>
+</ul>
+</section>
+
+<h2>Pets Applied for</h2>
+<ul>
+  <% @application.pets.each do |pet| %>
+ <br><%= link_to "#{pet.name}", "/pets/#{pet.id}" %>
+  <% if pet.application_pet(@application.id).status.nil? %>
+        <%= render partial: 'button', locals: {
+                                      url: "/admin/application_pets/#{@application.id}?pet_id=#{pet.id}&application_id=#{@application.id}&status=approved",
```

Comment:
You have your application_id twice here -- once as a path variable and another as a query parameter. You can just use one of these! You could also feel free to remove the `/admin` to the beginning since this is sort of an "internal" route, not one that the user sees. Another option is adding the pet ID directly to the path rather than a query parameter. like so: "/applications/:application_id/pets/:pet_id?status=approved" ---> still a RESTful path!